src/Symfony/Component/Validator/Mapping/GenericMetadata.php

```diff
@@ -231,13 +231,16 @@ public function getTraversalStrategy()
      *
      * Should not be used.
      *
+     * Implemented for backward compatibility with Symfony < 2.5.
+     *
      * @throws BadMethodCallException
      *
-     * @deprecated Implemented for backwards compatibility with Symfony < 2.5.
-     *             Will be removed in Symfony 3.0.
+     * @deprecated since version 2.5, to be removed in 3.0.
      */
     public function accept(ValidationVisitorInterface $visitor, $value, $group, $propertyPath)
     {
+        trigger_error('The '.__METHOD__.' is deprecated since version 2.5 and will be removed in 3.0.', E_USER_DEPRECATED);
```

Comment:
should be removed IMO as it throws a BadMethodCallException already because it is not part of the 2.5 API (we might want to improve the exception message though)